src/Microsoft.Security.Utilities.Core/RegexPattern.cs

```diff
@@ -38,13 +45,14 @@ public RegexPattern(string id,
         DetectionMetadata = patternMetadata;
         RotationPeriod = rotationPeriod;
         Signatures = signatures;
-        m_regexOptions = regexOptions;
+        RegexOptions = (regexOptions | RegexDefaults.DefaultOptions) & ~disabledRegexOptions;
```

Comment:
This is maybe unnecessarily complicated. I could also just ask the caller to `|` on to `DefaultOptions` if that's what they want. I reasoned this would be a bigger break because previously the options you passed would get augmented by the engine with the DefaultOptions so I was trying to effectively retain that behavior. But I wonder if anyone is actually using this argument now. If I was sure nobody was, I'd go for the simpler approach.

Reply:
I had a different thought, simply make this a nullable value. If null, prefer the default options, otherwise use the options provided by the caller in toto.

I do *not* believe anyone is using this argument currently, so have at it (in terms of reaching for greatest simplicity).

Reply:
Hey, because you are a scrupulous developer, I suggest you update your PR description to note the interesting subtleties in behavior that drove us landing on the nullable num arg.

Reply:
Done.